includes/class-sensei-quiz.php

```diff
@@ -1523,6 +1528,53 @@ public static function start_quiz_questions_loop() {
 		$sensei_question_loop['quiz_id']   = $quiz_id;
 	}
 
+	/**
+	 * Display quiz progress bar.
+	 *
+	 * @param int   $quiz_id quiz id.
+	 * @param array $pagination_settings pagination settings.
+	 * @since 3.15.0
+	 */
+	public static function display_progress_bar( $quiz_id, $pagination_settings ) {
+
+		$user_id             = get_current_user_id();
+		$lesson_id           = Sensei()->quiz->get_lesson_id( $quiz_id );
+		$all_questions_count = count( Sensei()->lesson->lesson_quiz_questions( $quiz_id, 'publish' ) );
```

Comment:
You could use the global `$sensei_question_loop['total']` to get the count instead of querying the questions again.

Reply:
https://github.com/Automattic/sensei/pull/4606/files#diff-964a2368a54a4fd7dd7d8dfe593a8f74e690d028590dc23dbb8d1d4e5b68577fR1673